### ISSUE TITLE
GetTextLength quirks to make note of

### ISSUE THREAD

**Issue by WishMakers0** [UltimaOmega/TouhouDanmakufuRemastered#3]
1) ASCII characters in text objects are treated as full-width JP characters internally - this means ObjText_GetTextLength as well as its variants will be higher than you expect.
2) GetTextLength and its variants also generate a text length based on number of characters ONLY, meaning each English character will be interpreted as monospace - as a result, the factor your length will be different will range anywhere from 1.5 to 2 times the length you expect.

Ideally, there should be a ObjText_GetTextLengthASCII function created (FEATURE REQUEST), as well as other functions necessary for handling other languages if time permits.
(This issue was found during the creation of Kaisendo's TriFocuser English patch)

**Comment by Akaricchi** [comment]
I am assuming that `ObjText_GetTextLength` is supposed to return the amount of characters in a string, and is not to be used to infer the on-screen width of the rendered text. Though gathering from what's written both here and on the DNF wiki, it sounds broken either way...

Here's my two cents:

Having different APIs for each language/encoding sounds like absurd. Ideally, your text abstraction should be language- and encoding-agnostic (especially encoding-agnostic!). The easiest way to do that is to represent text internally as a sequence of Unicode code points (in UCS-4 for example). Then the "length" of a text string is just the number of code points it contains. This is normally what you'd expect and want for text manipulation purposes, as each code point typically maps to one logical character, i.e. `length("Reimu") = 5` and `length("霊夢") == 2`. Of course, this is not very useful on its own, you'd need a basic set of string manipulation functions that understand these semantics. Then you can keep whatever danmakufu currently offers for compatibility, but deprecate it in favor of the unicode APIs.

**Comment by WishMakers0** [comment]
I agree with this - the point of keeping the existing functionality is purely for compatibility purposes (which is really the only reason why I suggested a different function for ASCII in the first place) but in general using Unicode as the standard makes more sense.  The plans for compatibility are not set in stone, so the existing functions may be depreciated in the future.
